examples/md-flexible/src/configuration/MDFlexConfig.cpp

```diff
@@ -547,12 +556,18 @@ void MDFlexConfig::loadParticlesFromCheckpoint(const size_t &rank, const size_t
     return;
   }
 
-  size_t checkpointCommunicatorSize{getNumPiecesInCheckpoint(filename)};
-  if (communicatorSize == checkpointCommunicatorSize) {
-    loadParticlesFromRankRecord(filename, rank, _particles);
-  } else {
-    for (size_t i = 0; i < checkpointCommunicatorSize; ++i) {
-      loadParticlesFromRankRecord(filename, i, _particles);
+  // If the checkpoint was written by a MPI parallel simulation the checkpoint is a pvtu file,
+  // that points to several vtu files, called pieces.
+  size_t numCheckpointPieces{getNumPiecesInCheckpoint(filename)};
+  // If the number of currently used ranks matches the number of pieces in the checkpoint,
+  // we assume that the domain layout matches and each rank only loads their own piece.
+  // If it doesn't match the overlap will be redistributed when particles are inserted into the AutoPas containers.
+  if (numRanks == numCheckpointPieces) {
+    loadParticlesFromRankRecord(filename, rank, particles);
+  } else if (rank == 0) {
+    // Otherwise, each rank 0 loads everything and distributes it later.
```

Comment:
I assume "each" is a typo:
```suggestion
    // Otherwise, rank 0 loads everything and distributes it later.
```